src/os/bluestore/BlueStore.cc

```diff
@@ -4321,8 +4371,73 @@ void BlueStore::_set_blob_size()
            << std::dec << dendl;
 }
 
+void BlueStore::_set_config_changed()
+{
+  std::lock_guard l(config_lock);
+  config_changed++;
+}
+
+uint32_t BlueStore::_get_config_changed()
+{
+  std::lock_guard l(config_lock);
+  return config_changed;
+}
+
+void BlueStore::_update_osd_memory_target_size()
+{
+  osd_memory_target = cct->_conf.get_val<Option::size_t>("osd_memory_target");
+  _set_config_changed();
+  dout(10) << __func__ << " osd_memory_target " << osd_memory_target << dendl;
+}
+
+void BlueStore::_update_cgroup_limit_ratio()
+{
+  if (_set_cgroup_limit_ratio()) {
+    _update_osd_memory_target_size();
+    _set_config_changed();
+  }
+}
+
+void BlueStore::_update_osd_memory_options()
+{
+  osd_memory_base = cct->_conf.get_val<Option::size_t>("osd_memory_base");
+  osd_memory_expected_fragmentation = cct->_conf.get_val<double>("osd_memory_expected_fragmentation");
+  osd_memory_cache_min = cct->_conf.get_val<Option::size_t>("osd_memory_cache_min");
+  _set_config_changed();
+  dout(10) << __func__
+           << " osd_memory_base " << osd_memory_base
+           << " osd_memory_expected_fragmentation " << osd_memory_expected_fragmentation
+           << " osd_memory_cache_min " << osd_memory_cache_min
+           << dendl;
+}
+
+bool BlueStore::_set_cgroup_limit_ratio()
```

Comment:
Looks like https://github.com/ceph/ceph/pull/29581 just removed this stuff from _set_cache_sizes(). Are you sure you want it back?

Reply:
Since the intent of #29581 is to remove the re-calculation of osd_memory_target based on the cgroup_ratio, there's no necessity for a config observer for this option anymore. I will remove the observer for this option.

Reply:
I have removed the observer for cgroup option.